File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int maxFrequencyElements(int[] nums) {
        Map<Integer, Integer> freqMap = new HashMap<>();
        int maxFreq = 0;

        for (int num : nums) {
            freqMap.put(num, freqMap.getOrDefault(num, 0) + 1);
            maxFreq = Math.max(maxFreq, freqMap.get(num));
        }

        int count = 0;
        for (int freq : freqMap.values()) {
            if (freq == maxFreq) {
                count++;
            }
        }

        return count * maxFreq;
    }
}

File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> u_map;
        int max_freq = 0;
        for(auto& x: nums) {
            u_map[x]++;
            max_freq = max(max_freq, u_map[x]);
        }
        int count = 0;
        for(auto& p: u_map) {
            if(p.second == max_freq)
                count += max_freq;
        }
        return count;
    }
};